File: backend/src/Enum/ContractType.php
<?php

namespace App\Enum;

enum ContractType: string
{
    case CDI = 'CDI';
    case CDD = 'CDD';
    case FREELANCE = 'FREELANCE';
    case INTERNSHIP = 'INTERNSHIP';
    case ALTERNANCE = 'ALTERNANCE';
    case OTHER = 'OTHER';
}

File: backend-java/src/main/java/com/jobdesk/domain/ContractType.java
package com.jobdesk.domain;

/**
 * Type de contrat. Les constantes == valeurs stockées en base,
 * donc mappé via @Enumerated(EnumType.STRING).
 */
public enum ContractType {
    CDI,
    CDD,
    FREELANCE,
    INTERNSHIP,
    ALTERNANCE,
    OTHER
}
